client/src/ingredients/browser-compatibility-table/browser-compatibility-rows.js

```diff
@@ -0,0 +1,226 @@
+import React from "react";
+import { BrowserSupportDetail } from "./browser-support-detail";
+import { BrowserSupportNotes } from "./browser-support-notes";
+
+function buildCompatibilityObject(compatibilityData, displayBrowsers) {
+  const features = {};
+
+  if (!!compatibilityData.__compat) {
+    const name = compatibilityData.__compat.mdn_url.split('/').pop();
+    features[name] = compatibilityData.__compat;
+    for (const compat in compatibilityData) {
+      if (compat !== "__compat" && !!compatibilityData[compat]["__compat"]) {
+        features[compat] = compatibilityData[compat]["__compat"];
+      }
+    }
+  }
+
+  return features;
+}
+
+function getVersionAdded(support) {
+  // When version compatibility is "unknown", it will be `undefined`.
+  // In that case, return `null` so we render the "?" block.
+  if (support === undefined) {
+    return null;
+  }
+
+  if (Array.isArray(support)) {
+    return support[0].version_added;
+  }
+
+  return support.version_added;
+}
+
+function getIndexNoteForBrowserDetail(indexNotes, browserDetailIndex) {
+  return indexNotes.find(indexNotes => indexNotes.index === browserDetailIndex);
+}
+
+function RenderBrowserSupportDetails({browserSupportDetails, rowIndex, indexNotes, currentNoteId, onNotesClick}) {
+  return browserSupportDetails.map((browserSupportDetail, detailIndex) =>
+    <BrowserSupportDetail
+      key={`${rowIndex}-${detailIndex}`}
```

Comment:
Try to avoid building keys from integers. It's extremely error-prone and can lead to very subtle bugs. 
Since Stumptown is a SPA, the BCD table might mount once and re-rendered with *different* `document.browser_compatibility` structs and this can lead to incorrect re-renders. 
Here's a good example: https://medium.com/@vraa/why-using-an-index-as-key-in-react-is-probably-a-bad-idea-7543de68b17c

Try to come up with something unique instead. To be honest I don't really know what `browserSupportDetail` is but perhaps you can use `key={this.computeDistinctKey(browserSupportDetail)}` and write a little function that does something like...

```javascript
computeDistinctKey(detail) {
  return `${detail.browser}:${detail.support}:${detail.version_added}`
}
```